metered/src/hdr_histogram.rs

```diff
@@ -89,25 +89,57 @@ impl Serialize for HdrHistogram {
         S: Serializer,
     {
         let hdr = &self.histo;
-        let ile = |v| hdr.value_at_percentile(v);
+
+        /// A percentile of this histogram - for supporting serialisers this will
+        /// ignore the key (such as `90%ile`) and instead add a dimension to the
+        /// metrics (such as `quantile=0.9`).
+        macro_rules! ile {
+            ($e:expr) => {
+                &NewTypeAlias(concat!("!|quantile=", $e), hdr.value_at_quantile($e))
+            }
+        }
+
+        /// A 'qualified' metric name - for supporting serialisers this will prepend
+        /// the metric name to this key, outputting `response_time_count`, for example
+        /// rather than just `count`.
+        macro_rules! qual {
+            ($e:expr) => {
+                &NewTypeAlias("<|", $e)
+            }
+        }
+
         use serde::ser::SerializeMap;
 
         let mut tup = serializer.serialize_map(Some(10))?;
-
-        tup.serialize_entry("samples", &hdr.len())?;
-        tup.serialize_entry("min", &hdr.min())?;
-        tup.serialize_entry("max", &hdr.max())?;
-        tup.serialize_entry("mean", &hdr.mean())?;
-        tup.serialize_entry("stdev", &hdr.stdev())?;
-        tup.serialize_entry("90%ile", &ile(90.0))?;
-        tup.serialize_entry("95%ile", &ile(95.0))?;
-        tup.serialize_entry("99%ile", &ile(99.0))?;
-        tup.serialize_entry("99.9%ile", &ile(99.9))?;
-        tup.serialize_entry("99.99%ile", &ile(99.99))?;
+        tup.serialize_entry("samples", qual!(hdr.len()))?;
+        tup.serialize_entry("min", qual!(hdr.min()))?;
+        tup.serialize_entry("max", qual!(hdr.max()))?;
+        tup.serialize_entry("mean", qual!(hdr.mean()))?;
+        tup.serialize_entry("stdev", qual!(hdr.stdev()))?;
+        tup.serialize_entry("90%ile", ile!(0.9))?;
+        tup.serialize_entry("95%ile", ile!(0.95))?;
+        tup.serialize_entry("99%ile", ile!(0.99))?;
+        tup.serialize_entry("99.9%ile", ile!(0.999))?;
+        tup.serialize_entry("99.99%ile", ile!(0.9999))?;
         tup.end()
     }
 }
 
+/// This is a mocked 'newtype' (eg. `A(u64)`) that instead allows us to
+/// define our own type name that doesn't have to abide by Rust's constraints
+/// on type names. This allows us to do some manipulation of our metrics,
+/// allowing us to add dimensionality to our metrics via key=value pairs, or
+/// key manipulation on serialisers that support it.
+struct NewTypeAlias<T: Serialize>(&'static str, T);
```

Comment:
Can we maybe call this `MetricAlias` ?

Reply:
Yep, not a problem. This is done in [fada99..807134](https://github.com/magnet/metered-rs/compare/fada9953c90970bcb02300a7f3dde83e70fe4dbe..80713412cc5c11f315a05c7562934fd6f89eb189)